<launch>
  <arg name="world_frame" default="ugr/car_odom"/>
  <arg name="base_link_frame" default="ugr/car_base_link"/>
  <arg name="slam_base_link_frame" default="ugr/slam_base_link"/>
  <arg name="lidar_frame" default="os_sensor"/>

  <node pkg="slam_graphslam" output="screen" type="graphslam" name="graphslam" respawn="False">
    # Parameters
    <param name="world_frame" value="$(arg world_frame)" />
    <param name="base_link_frame" value="$(arg base_link_frame)" />
    <param name="slam_base_link_frame" value="$(arg slam_base_link_frame)" />
    <param name="lidar_frame" value="$(arg lidar_frame)" />

    <param name="target_rate" value="0.5"/>
    <param name="publish_rate" value="3.0"/>
    <param name="synchronous" value="false"/>

    <param name="max_iterations" value="10"/>

    # Input/output remapping
    <remap from="/input/observations" to="ugr/car/observations/lidar" />

    <remap from="/output/observations" to="/ugr/car/observations/slam" />
    <remap from="/output/odom" to="/ugr/car/odometry/filtered/slam" />
  </node>
</launch>
